apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -796,6 +796,16 @@ func (p *ProvisionerAPI) GetContainerInterfaceInfo(args params.Entities) (
 	return p.prepareOrGetContainerInterfaceInfo(args, true)
 }
 
+// Machine is an indirection for use in container provisioning.
+//go:generate mockgen -package provisioner_test -destination machine_mock_test.go github.com/juju/juju/apiserver/facades/agent/provisioner Machine
+//go:generate mockgen -package provisioner_test -destination linklayerdevice_mock_test.go github.com/juju/juju/network/containerizer LinkLayerDevice
+type Machine interface {
+	containerizer.Container
```

Comment:
I generally don't prefer interfaces to embed interfaces from an external package, unless there is a strong reason. Otherwise its hard to track down what the actual interface is.

Reply:
In this case, we are passing instances of it through to the containerizer bridge policy logic.

This means that we would need to replicate the test-only method declarations and shim methods like `Raw` locally, which I think would diminish its cogency in provisioner.